File: src/main/java/com/evolutions/database/autoconfigure/DatabaseEvolutionsProperties.java
package com.evolutions.database.autoconfigure;

import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * Configuration properties for Database Evolutions.
 *
 * @author Gabriel Oliveira
 */
@ConfigurationProperties(prefix = "play.evolutions")
public class DatabaseEvolutionsProperties {

    private boolean enable = true;

    private String schema = "";

    private boolean autocommit = true;

    private boolean useLocks = false;

    private boolean autoApply = false;

    private boolean autoApplyDowns = false;

    private boolean skipApplyDownsOnly = false;

    public boolean isEnable() {
        return enable;
    }

    public void setEnable(boolean enable) {
        this.enable = enable;
    }

    public String getSchema() {
        return schema;
    }

    public void setSchema(String schema) {
        this.schema = schema;
    }

    public boolean isAutocommit() {
        return autocommit;
    }

    public void setAutocommit(boolean autocommit) {
        this.autocommit = autocommit;
    }

    public boolean isUseLocks() {
        return useLocks;
    }

    public void setUseLocks(boolean useLocks) {
        this.useLocks = useLocks;
    }

    public boolean isAutoApply() {
        return autoApply;
    }

    public void setAutoApply(boolean autoApply) {
        this.autoApply = autoApply;
    }

    public boolean isAutoApplyDowns() {
        return autoApplyDowns;
    }

    public void setAutoApplyDowns(boolean autoApplyDowns) {
        this.autoApplyDowns = autoApplyDowns;
    }

    public boolean isSkipApplyDownsOnly() {
        return skipApplyDownsOnly;
    }

    public void setSkipApplyDownsOnly(boolean skipApplyDownsOnly) {
        this.skipApplyDownsOnly = skipApplyDownsOnly;
    }
}

File: autoconfigure/src/main/scala/com/evolutions/database/autoconfigure/DatabaseEvolutionsProperties.scala
package com.evolutions.database.autoconfigure

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * Configuration properties for Database Evolutions.
 *
 * @author Gabriel Oliveira
 */
@ConfigurationProperties(prefix = "play.evolutions")
class DatabaseEvolutionsProperties {

  private var enable = true

  private var schema = ""

  private var autocommit = true

  private var useLocks = false

  private var autoApply = false

  private var autoApplyDowns = false

  private var skipApplyDownsOnly = false

  private var mode = "dev"

  private var path = "evolutions"

  def isEnable: Boolean = enable

  def setEnable(enable: Boolean): Unit = {
    this.enable = enable
  }

  def getSchema: String = schema

  def setSchema(schema: String): Unit = {
    this.schema = schema
  }

  def isAutocommit: Boolean = autocommit

  def setAutocommit(autocommit: Boolean): Unit = {
    this.autocommit = autocommit
  }

  def isUseLocks: Boolean = useLocks

  def setUseLocks(useLocks: Boolean): Unit = {
    this.useLocks = useLocks
  }

  def isAutoApply: Boolean = autoApply

  def setAutoApply(autoApply: Boolean): Unit = {
    this.autoApply = autoApply
  }

  def isAutoApplyDowns: Boolean = autoApplyDowns

  def setAutoApplyDowns(autoApplyDowns: Boolean): Unit = {
    this.autoApplyDowns = autoApplyDowns
  }

  def isSkipApplyDownsOnly: Boolean = skipApplyDownsOnly

  def setSkipApplyDownsOnly(skipApplyDownsOnly: Boolean): Unit = {
    this.skipApplyDownsOnly = skipApplyDownsOnly
  }

  def getMode: String = mode

  def setMode(mode: String): Unit = {
    this.mode = mode
  }

  def getPath: String = path

  def setPath(path: String): Unit = {
    this.path = path
  }

}
